### PR TITLE
Recommend pipeline

### PR DESCRIPTION
**Pull request by lyriccoder**
Fixing https://github.com/yegor256/aibolit/issues/246

Do not review, do not merge

@0crat  out

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#248](https://github.com/yegor256/aibolit/issues/248) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/d556c9d0-dcc6-45b1-884c-45e1981d7c37); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: aibolit/ml_pipeline/ml_pipeline.py [hunk 1]
```diff
@@ -102,17 +103,25 @@ def train_process(model_folder=None):
 
         dataset = Dataset(only_patterns)
         dataset.preprocess_file()
+        cwd = Path(os.getcwd())
+        features_conf = {
+            "features_order": dataset.feature_order,
+            "patterns_only": only_patterns
+        }
+        with open(r'D:\git\aibolit\aibolit\binary_files\features_order.json', 'w') as fid:
```
Comment (acheshkov) [inline aibolit/ml_pipeline/ml_pipeline.py:111]: what is this hardcoded path?

File: aibolit/model/model.py [hunk 2]
```diff
@@ -61,9 +25,9 @@ def preprocess_file(
             scale=False,
             **kwargs):
 
-        # df = pd.read_csv(r'D:\git\aibolit\scripts\target\dataset.csv')
-        print(Path(os.getcwd(), 'target', 'dataset.csv'))
-        df = pd.read_csv(str(Path(os.getcwd(), 'target', 'dataset.csv')))
+        df = pd.read_csv(r'D:\git\aibolit\scripts\dataset.csv')
```
Comment (acheshkov) [inline aibolit/model/model.py:28]: why we hardcode this path?

File: aibolit/ml_pipeline/ml_pipeline.py (hunk 1 above)
Comment (lyriccoder) [reply aibolit/ml_pipeline/ml_pipeline.py:111]: it's just a draft, we will fix it.

**Comment by fanifieiev** [comment]
@lyriccoder What is the status of this PR ? Is it ready for review?

**Comment by lyriccoder** [comment]
No, it's not ready, I will notify @acheshkov, we are discussing it.

**Comment by 0crat** [comment]
Are you speaking to me or about me [here](https://github.com/yegor256/aibolit/issues/248#issuecomment-620683439); you must always start your message with my name if you want to address it to me, see [§1](http://www.zerocracy.com/policy.html#1)

File: aibolit/model/model.py [hunk 3]
```diff
@@ -61,8 +25,6 @@ def preprocess_file(
             scale=False,
             **kwargs):
 
-        # df = pd.read_csv(r'D:\git\aibolit\scripts\target\dataset.csv')
-        print(Path(os.getcwd(), 'target', 'dataset.csv'))
         df = pd.read_csv(str(Path(os.getcwd(), 'target', 'dataset.csv')))
```
Comment (acheshkov) [inline aibolit/model/model.py:28]: we should pass the filename of dataset as parameter.

File: aibolit/model/model.py [hunk 4]
```diff
@@ -61,21 +23,19 @@ def preprocess_file(
...
-        df = pd.read_csv(str(Path(os.getcwd(), 'target', 'dataset.csv')))
+        df = pd.read_csv(Config.dataset_file())
         df = df[~df["filename"].str.lower().str.contains("test")]
 
         if self.do_rename_columns:
             p_codes = \
-                [x['code'] for x in CONFIG['patterns']] \
-                + ['lines' + x['code'] for x in CONFIG['patterns']]
-            m_codes = [x['code'] for x in CONFIG['metrics']]
+                [x['code'] for x in Config.get_patterns_config()['patterns']] \
```
Comment (fanifieiev) [inline aibolit/model/model.py:31]: @lyriccoder Why not assigning `Config.get_patterns_config()` to local variable once and use the variable which would make the code cleaner?
```
config = Config.get_patterns_config()
....
[x['code'] for x in config['patterns']]
```

**Comment by 0crat** [comment]
@fanifieiev[/z](https://www.0crat.com/u/fanifieiev) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by acheshkov** [comment]
@rultor merge

**Comment by acheshkov** [comment]
@rultor merge

**Comment by acheshkov** [comment]
@rultor merge

**Comment by 0crat** [comment]
Code review was too long (8 days), architects (@acheshkov) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Pull request merged**